templates/_components/_image.twig

```diff
@@ -3,13 +3,154 @@
+	@param   class                {string}                  Class attribute for the <img> tag
+	@param   alt                  {string}                  Alt attribute for the <img> tag
+	@param   lazyload             {bool}                    Whether to lazyload images (default: true)
+	@param   background           {string}                  Class name for background image
+	@param   webp                 {bool}                    Whether to generate WebP versions (default: true)
+
+	Example of simple usage:
+  --------------
+	```
+  <figure>
+    {% include '_components/_image' with { params: {
+      asset: image,
+      class: 'profile-pic',
+      preset: 'avatar',
+    } } %}
+	</figure>
+  ```
+
+	Example of usage with breakpoint asset:
+	------------------------------------
+	```
+	<figure>
+	{% include '_components/_image' with { params: {
+    asset: image,
+    breakpoint_assets: {
+     '--lg-plus': imageLarge,
+		 '--md-plus': imageMedium,
+		 '--sm-plus': imageSmall,
+	  },
+  } } %}
+	</figure>
+	```
+	
+	Set `presets`(a set of image transformation configurations) variable values in `_image` component.
+	Define the desired preset values in the image component as shown in the examples below.
+
+	Preset options:
+	---------------
+	@option   media          {string}      Any valid media query
+	@option   breakpoints    {array}       Array of transform values for different view breakpoints
+	@option   sizes          {array}       Array of size based transform values
+	@option   transform      {array}       Array of aspect and position related transform values
+	@option   guide          {string}      Any valid media query with size
+	
+	Preset examples:
+	---------------
+
+	1) Simple size-based transforms (can be width, or height — everything that the Imager plugin supports)
+		```
+		{% set presets = {
+			  avatar: {
+			    sizes: [
+			      { width: 20 },
+			      { width: 30 },
+			      { width: 40 },
+				   ],
+				 },	
+	  } %}
+		```
+
+		HTML Output:
+		`<img src=”small.jpg” srcset=”small.jpg 20w, medium.jpg 30w, large.jpg 40w”>`
+	
+	2) Size based transform with option to crop images
+    ```
+    {% set presets = {
+      blog: {
+        sizes: [
+          { width: 400 },
+          { width: 800 },
+        ],
+        transform: { mode: 'crop', ratio: 1.7 },
+      },	
+    } %}
+    ```
+
+		HTML Output:
+		`<img src=”small.jpg” srcset=”small.jpg 400w, large.jpg 800w”>`
+
+	3) Size based transform with different sizes based on render area sizes 
+    ```
+    {% set presets = {
+      logo: {
+        sizes: [
+          { width: 150 },
+          { width: 300 },
+        ],
+        guide: '--md-plus 5rem, --lg-plus 8rem, 3.55rem',
+      },
+    } %}
+    ```
+
+		HTML Output:
+		`<img src=”small.png” sizes=”(min-width:1188px) 5rem, (min-width:700px) 4rem, 3.55rem” srcset=”small.png 150w, medium.png 300w”>`
+
+	4) Art-directed images
+    ```
+    {% set queries = {
```

Comment:
This should be in its own sub-section and explained just like presets.

Reply:
Added.